pkg/kernel/kernel.go

```diff
@@ -69,7 +80,8 @@ func New(ctx context.Context, config cfg.Config, logger log.Logger, options ...O
 		started:   conc.NewPoisonedLock(),
 
 		killTimeout: time.Second * 10,
-		forceExit:   os.Exit,
+		exitCode:    ExitCodeOk,
```

Comment:
By default the exit code should be an error in my opinion and only be changed to success once we complete running the app - otherwise, if you panic, you would return an exit success, right?

Reply:
changed